tests/csapi/sync_archive_test.go

```diff
@@ -190,43 +190,106 @@ func TestArchivedRoomsHistory(t *testing.T) {
 		},
 	})
 
-	exhaustiveResCheck := func(result gjson.Result, origin string) {
-		roomRes := result.Get("rooms.leave." + client.GjsonEscape(roomID))
+	t.Run("timeline has events", func(t *testing.T) {
```

Comment:
Can we comment on what each of these subtests are testing here please? (I know this one toy didn't add, but since you're here...)

Reply:
Have added a couple of comments - are they clear enough?